addons/web/static/src/core/macro.js

```diff
@@ -174,24 +174,26 @@ export class MacroEngine {
         };
         this.observer = new MutationObserver(this.delayedCheck.bind(this));
         this.iframeObserver = new MutationObserver(() => {
```

Comment:
We could benefit from the arguments passed to this callback (see https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver).

With the current version, you will re-observe the iframe each time a mutation is triggered somewhere in the main document. This callback takes a mutationList is params, in it you'll find the addedNodes. You can use it to obtain the iframes, instead of looking into the DOM.

Reply:
What do you think about that ?